### PR TITLE
Isaacrae/sc 29437/observability kit logger package

### PR DESCRIPTION
**Pull request by irae0385**

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	case strings.ToLower(EnvProd):
+		opts.Level = slog.LevelError
+	}
+
+	handler := sHandler{handler: slog.NewJSONHandler(w, &opts)}
+	Logger = slog.New(&handler)
+}
+
+// LogRequestIDMiddleware middleware for logging using requestID middleware from Chi
+func LogRequestIDMiddleware(next http.Handler) http.Handler {
```
Comment (cheddarnaz) [inline logger/logger.go:74]: Pull this out into a middleware.go file within the logger package. Other middleware can also live there.

**Review by cheddarnaz — CHANGES_REQUESTED** [review]

File: logger/logger.go (hunk 1 above)
Comment (irae0385) [reply logger/logger.go]: ended up pulling the handler into it's own file to.

File: logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	// no default required as log level is by default Info
+	switch strings.ToLower(envLevel) {
+	case strings.ToLower(EnvDev):
+		opts.Level = slog.LevelDebug
+	case strings.ToLower(EnvStaging): // eventually Dev environment would be level info too
+		opts.Level = slog.LevelInfo
+	case strings.ToLower(EnvProd):
+		opts.Level = slog.LevelError
```
Comment (cheddarnaz) [inline logger/logger.go:37]: Can you make one more change here? Change this to `slog.LevelWarn`

**Review by cheddarnaz — CHANGES_REQUESTED** [review]

File: logger/logger.go (hunk 2 above)
Comment (irae0385) [reply logger/logger.go]: done

**Review by cheddarnaz — APPROVED** [review]
lgtm

**Pull request merged**